sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,291 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private ChangeTrackingStringList? _allowedHeaderNames;
+    private ChangeTrackingStringList? _allowedQueryParameters;
+
+    private static string[] DefaultAllowedHeaderNames { get; } = [
```

Comment:
I updated the default values to avoid allocations in the default sanitizer path. I left the rest as-is. I might try to experiment a bit more with changing ChangeTrackingStringList to be ChangeTrackingStringHashSet and the collection properties (allowed headers and allowed queries) to ICollection instead of IList.